compiler/AST/type.cpp

```diff
@@ -1612,6 +1623,13 @@ bool isClassLikeOrPtr(Type* t) {
                             t == dtCFnPtr);
 }
 
+bool isCPtrConstChar(Type* t) {
+  if (auto dct = getDataClassType(t->symbol))
+    return dct->typeInfo() == dt_c_char &&
+            t->symbol->hasFlag(FLAG_C_PTRCONST_CLASS);
```

Comment:
IMO it makes more sense to check the flag before calling `getDataClassType` but it should not affect correctness